nanoFramework.Json.Test/JsonUnitTests.cs

```diff
@@ -846,6 +846,69 @@ public void SerializeObjectAsAProperty()
             Assert.Equal(correctValue, json, "Serialize object as property fails");
         }
 
+        [TestMethod]
+        public void DeserializeSingleTypesClassDeserialization()
+        {
+            var json = "{\"OneByte\":42,\"OneSByte\":-42,\"OneInt16\":1234,\"OneUInt16\":5678,\"OneInt32\":-789012,\"OneUInt32\":78912,\"OneInt64\":-1234567,\"OneUInt64\":1234567,\"OneSingle\":34.45,\"OneDouble\":45678.23,\"OneBoolean\":true,\"TwoBoolean\":false}";
+            var deser = JsonConvert.DeserializeObject(json, typeof(SingleTypesClassDeserialization)) as SingleTypesClassDeserialization;
+            Assert.Equal((byte)42, deser.OneByte, "Byte");
+            Assert.Equal((sbyte)-42, deser.OneSByte, "SByte");
+            Assert.Equal((short)1234, deser.OneInt16, "Int16");
+            Assert.Equal((ushort)5678, deser.OneUInt16, "UInt16");
+            Assert.Equal(-789012, deser.OneInt32, "Int32");
+            Assert.Equal((uint)78912, deser.OneUInt32, "UInt32");
+            Assert.Equal((long)-1234567, deser.OneInt64, "Int64");
+            Assert.Equal((ulong)1234567, deser.OneUInt64, "UInt64");
+            Assert.Equal((float)34.45, deser.OneSingle, "Single");
+            Assert.Equal((double)45678.23, deser.OneDouble, "Double");
+            Assert.True(deser.OneBoolean,"Boolean true");
+            Assert.False(deser.TwoBoolean,"Boolean false");
+        }
+
+        [TestMethod]
+        public void DeserializeArrayToDeserialize()
+        {
+            ArrayToDeserialize obj0 = new() { Prop1 = 1, Prop2 = "prop2", Prop3 = true, Prop4 = 67890123 };
+            ArrayToDeserialize obj1 = new() { Prop1 = -42, Prop2 = "second2", Prop3 = false, Prop4 = 123456 };
+            ArrayToDeserialize[] array = new[] { obj0, obj1 };
+            var json = JsonConvert.SerializeObject(array);
+
+            var deser = JsonConvert.DeserializeObject(json, typeof(ArrayToDeserialize[])) as ArrayToDeserialize[];
+
+            Assert.Equal(deser.Length ,array.Length, "Aray length");
```

Comment:
I don't know how picky you accept PR comments, but there's a typo : "Array length", and the first parameter should have the space after the comma.

Reply:
ctrl + k + d was still missing on the file. Done. Thx